lib/pbench/agent/tool_meister.py

```diff
@@ -1285,6 +1285,50 @@ def stop_tools(self, data):
             self._send_client_status("success")
         return failures
 
+    def _create_tar(
+        self,
+        target_dir: Path,
+        tar_file: Path,
+        cwd: Path = None,
+        ignore_warning: bool = False,
+    ) -> subprocess.CompletedProcess:
+        """
+        Creates a tar file at a given tar file path.
+        """
+        tar_args = [
+            self.tar_path,
+            "--create",
+            "--xz",
+            "--force-local",
+            f"--file={tar_file}",
+            target_dir,
+        ]
+        if ignore_warning:
+            tar_args.insert(2, "--warning=none")
+
+        try:
+            # Invoke tar directly for efficiency.
+            cp = subprocess.run(
+                tar_args,
+                cwd=cwd if cwd else target_dir.parent,
+                stdin=None,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
+            if cp.returncode != 0:
+                self.logger.warning(
+                    "Failed to create tar ball with arguments %s; Error: %s",
+                    str(tar_args),
+                    cp.stdout.decode("utf-8"),
+                )
+        except Exception:
+            self.logger.exception(
+                "Failed to create tar ball with arguments '%s'", str(tar_args)
+            )
+        else:
+            return cp
+        return None
```

Comment:
I don't know how everyone feels about this, but my preferences are to minimize the code inside the `try` block and to return immediately when we fail.  So, I would write this blob like:
```
        except Exception:
            self.logger.exception(
                "Failed to create tar ball with arguments '%s'", " ".join(tar_args)
            )
            return None

        if cp.returncode != 0:
            self.logger.warning(
                "Failed to create tar ball with arguments %s; Error: %s",
                " ".join(tar_args),
                cp.stdout.decode("utf-8"),
            )

        return cp
```

Reply:
This is the client side of a distributed system. Yeah, we always have to deal with the possibility of total nuclear disaster with timeouts and such, but we shouldn't ever just casually drop out without trying to maintain the protocol. We should at least stick a message in a bottle and toss it overboard as we sink ...